mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -6507,10 +6508,118 @@ struct ThreadwiseCopyRewritePattern
         SmallVector<Value, 8> destLowerIndices =
             layeredDestIndices[layeredDestIndices.size() - 1];
 
+        // we do not  implement the logic of toEmitOOBStoreCheck yet
+        bool toEmitOOBStoreCheckLogic = false;
+        SmallVector<unsigned, 2> oobStoreCheckDims;
+        // if (destLowerIndices.size() == 5) {
+        // for (unsigned iter = 0; iter < destTransform.getNumResults();
+        //     ++iter) {
+        // this code is for testing ,check dim 2
+        // if (iter == 2) {
+        //  toEmitOOBStoreCheckLogic = true;
+        //  oobStoreCheckDims.push_back(iter);
+        // }
+        //}
+        //}
+
         // Store to dest.
         // Issue scalar store.
-        b.create<StoreOp>(loc, convertedScalarValue, op.dest(),
-                          destLowerIndices);
+        if (toEmitOOBStoreCheckLogic) {
+          SmallVector<Value, 8> destLowerStoreOP;
+          SmallVector<Value, 8> destLowerStoreOOBOp;
+
+          auto dstIndex0 = b.create<IndexCastOp>(loc, destLowerIndices[0],
+                                                 b.getIntegerType(32));
+          auto dstIndex1 = b.create<IndexCastOp>(loc, destLowerIndices[1],
+                                                 b.getIntegerType(32));
+          auto dstIndex2 = b.create<IndexCastOp>(loc, destLowerIndices[2],
+                                                 b.getIntegerType(32));
+          auto dstIndex3 = b.create<IndexCastOp>(loc, destLowerIndices[3],
+                                                 b.getIntegerType(32));
+          auto dstIndex4 = b.create<IndexCastOp>(loc, destLowerIndices[4],
+                                                 b.getIntegerType(32));
+
+          destLowerStoreOP.push_back(dstIndex0);
+          destLowerStoreOP.push_back(dstIndex1);
+          destLowerStoreOP.push_back(dstIndex2);
+          destLowerStoreOP.push_back(dstIndex3);
+          destLowerStoreOP.push_back(dstIndex4);
+          destLowerStoreOOBOp = destLowerStoreOP;
+          // 2G ,INT MAX Value = 2147483647, use 2147483648 as offset and buffer
+          // store do nothing
+          const int twoGB = 2147483647;
```

Comment:
`twoGB` constant value is used between both `threadwise_copy` and `threadwise_copy_v2` so it's better to extract them to the beginning of the file.